### PR TITLE
07/10 branch

### PR DESCRIPTION
**Pull request by KsushaDk**
a lot of changes for you :)

### CODE REVIEW

File: src/pages/InterviewPage/InterviewPage.jsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+						/>
+					</div>
+					<div className="content__head_center">
+						<h2 className="title_s">Опрос: {interview.title}</h2>
+					</div>
+
+					<div className="content__body_center">
+						{interview.questions.map((question, index) => (
+							<div className="content__body_item_center" key={question.id}>
+								{interview.queries.requiredFields && question.required && (
```
Comment (maryia-kabash) [inline src/pages/InterviewPage/InterviewPage.jsx:149]: If you have long access to object properties, you can destruct it to make more readable - 
`const {requiredFields, progressBar, etc.} =  interview.queries`

File: src/utils/checkQueryForQuestion.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export const checkQueryForQuestion = (queries, key) => {
+	const checkQuery = queries.find(
+		(query) => query.key === key && query.checked === true
+	);
+
+	if (checkQuery) {
+		return true;
```
Comment (maryia-kabash) [inline src/utils/checkQueryForQuestion.js:7]: return !!checkQuery

File: src/utils/checkRequiredField.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const checkRequeredField = (questions) =>
+	questions
+		.map((question) => {
```
Comment (maryia-kabash) [inline src/utils/checkRequiredField.js:3]: you can just filter it, I think

File: src/utils/getOptionType.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Option } from '../constants/OptionType';
+
+export const getOptionType = (type) => {
+	let optionType = '';
```
Comment (maryia-kabash) [inline src/utils/getOptionType.js:4]: I guess Option.Text and other are numbers, so if used in props, may cause PropTypes warning

File: src/components/ui/progressbar/ProgressBar.scss [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+	&-fill {
+		position: absolute;
+		background: $primary-red;
+		height: 100%;
+		display: block;
+		width: 0;
+		border-radius: 4px;
+		-webkit-transition: width 0.8s ease;
+		-moz-transition: width 0.8s ease;
```
[the remark this answers is not in the document]
Comment (KsushaDk) [reply src/components/ui/progressbar/ProgressBar.scss:36]: post-css

File: src/components/QuestionTypeList/QuestionTypeList.jsx [hunk 6]
```diff
@@ -1,16 +1,24 @@
...
 	<div className="interview__settings">
 		<h3 className="title_xs">Тип вопроса</h3>
 		<ul className="settings__list" onClick={handleAddQuestion} role="menu">
-			{questionTypeList.map((questionType) => (
+			{questionTypeList.map((questionType, index) => (
 				<li
 					className="settings__list_option"
 					name={questionType.type}
+					id={questionType.id}
+					index={index}
```
[the remark this answers is not in the document]
Comment (KsushaDk) [reply src/components/QuestionTypeList/QuestionTypeList.jsx:15]: I use it for a11y: to handle arrow up and arrow down

File: src/components/InterviewQueryList/InterviewQueryList.jsx [hunk 7]
```diff
@@ -1,18 +1,25 @@
...
 import { CheckboxInput } from 'Components/ui/input/CheckboxInput/CheckboxInput';
+import { selectElByKeyDown } from 'Utils/selectElByKeyDown';
+import { propTypesConst } from 'Constants/propTypesConst';
 
 export const InterviewQueryList = ({ queries, handleChangeQuery }) => (
 	<div className="interview__settings">
 		<h3 className="title_xs">Параметры опроса</h3>
 		<ul className="settings__list" role="menu">
-			{queries.map((query) => (
+			{queries.map((query, index) => (
```
[the remark this answers is not in the document]
Comment (KsushaDk) [reply src/components/InterviewQueryList/InterviewQueryList.jsx:11]: I use it for a11y: to handle arrow up and arrow down

**Pull request merged**